avocado/utils/build.py

```diff
@@ -63,3 +63,52 @@ def make(path, make='make', env=None, extra_args='', ignore_status=False,
                                   allow_output_check=allow_output_check)
     os.chdir(cwd)
     return make_process
+
+# Dummy function : use process.run #
+
+
+def run_make(path, make='make', env=None, extra_args='', ignore_status=False,
+             allow_output_check='none'):
+    """
+    Run make, adding MAKEOPTS to the list of options.
+
+    :param make: what make command name to use.
+    :param env: dictionary with environment variables to be set before
+                calling make (e.g.: CFLAGS).
+    :param extra: extra command line arguments to pass to make.
+    :param allow_output_check: Whether to log the command stream outputs
+                               (stdout and stderr) of the make process in
+                               the test stream files. Valid values: 'stdout',
+                               for allowing only standard output, 'stderr',
+                               to allow only standard error, 'all',
+                               to allow both standard output and error,
+                               and 'none', to allow none to be
+                               recorded (default). The default here is
+                               'none', because usually we don't want
+                               to use the compilation output as a reference
+                               in tests.
+    :type allow_output_check: str
+    """
+    cwd = os.getcwd()
+    os.chdir(path)
+    cmd = make
+
+    # Set default number of jobs as ncpus + 1
+    if "-j" not in os.environ.get("MAKEFLAGS", ""):
+        jobs = multiprocessing.cpu_count() + 1
+        if not env:
+            env = {"MAKEFLAGS": "-j%s" % jobs}
+        elif "-j" not in env:
+            env["MAKEFLAGS"] = "-j%s" % jobs
+
+    makeopts = os.environ.get('MAKEOPTS', '')
+    if makeopts:
+        cmd += ' %s' % makeopts
+    if extra_args:
+        cmd += ' %s' % extra_args
+    make_process = process.run(cmd,
+                               env=env,
+                               ignore_status=ignore_status,
+                               allow_output_check=allow_output_check)
+    os.chdir(cwd)
+    return make_process
```

Comment:
Hello @PraveenPenguin, the `run_make` is more generic version of `make`, so instead of copy&paste you should rename the `make` to `run_make` and use the `process.run` there. Then you should add `make` function, which calls `run_make` and reports the `make_process.exit_status`, so it looks like this:

```python
def run_make(...):
    ...
    return make_process

def make(...):
    return run_make(...).exit_status
```

Last but not least, I'd not call the `run_make` a dummy function, it's a super-valid first-class function, so please remove the `dummy` comment.